### PR TITLE
[frontend] Bulk create relationship in container context (#7443)

### LINKED ISSUE #7443: POC on Bulk creation of SDO/SCO with Copy/Pasting: bulk create relationship
## Use case

Subset of https://github.com/OpenCTI-Platform/opencti/issues/4352

* Add a new dialog in the knowledge tab of Intrusion-Set (other places in platform to come in follow up PRs)
* Allow to create relations to SCO and SDO that already exists in the patform and matches the relation type.
* Everything on frontend side only

### PR DESCRIPTION
**Pull request by ValentinBouzinFiligran**
### Proposed changes

UNDER FEATURE FLAG: 4352_BULK_RELATIONS

* Add a new dialog in the knowledge tab of Intrusion-Set (other places in platform to come in follow up PRs)
* Allow to create relations to SCO and SDO that already exists in the patform and matches the relation type.

Everything is done in frontend as per brainstorm descision.

### Related issues

* https://github.com/OpenCTI-Platform/opencti/issues/7443

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality
- [x] I wrote test cases for the relevant uses case (coverage and e2e)
- [x] I added/update the relevant documentation (either on github or on notion)
- [x] Where necessary I refactored code to improve the overall quality

### CODE REVIEW

File: opencti-platform/opencti-graphql/src/utils/filtering/filtering-utils.ts [hunk 1]
```diff
@@ -264,7 +264,7 @@ export const checkAndConvertFilters = (filterGroup: FilterGroup | null | undefin
         }
       });
       if (incorrectKeys.length > 0) {
-        throw UnsupportedError('incorrect filter keys not existing in any schema definition', { keys: incorrectKeys });
+        throw UnsupportedError('incorrect filter keys not existing in any schema definition', { keys: incorrectKeys, avalaible: availableKeys });
```
Comment (aHenryJard) [inline opencti-platform/opencti-graphql/src/utils/filtering/filtering-utils.ts]:
```suggestion
        throw UnsupportedError('incorrect filter keys not existing in any schema definition', { keys: incorrectKeys });
```

File: opencti-platform/opencti-graphql/src/generated/graphql.ts [hunk 2]
```diff
@@ -17793,6 +17793,7 @@ export type Query = {
   backgroundTask?: Maybe<BackgroundTask>;
   backgroundTasks?: Maybe<BackgroundTaskConnection>;
   bookmarks?: Maybe<StixDomainObjectConnection>;
+  bulkSearch?: Maybe<StixCoreObjectConnection>;
```
Comment (aHenryJard) [inline opencti-platform/opencti-graphql/src/generated/graphql.ts:17796]:
```suggestion
```

File: opencti-platform/opencti-graphql/src/generated/graphql.ts [hunk 3]
```diff
@@ -18211,6 +18212,17 @@ export type QueryBookmarksArgs = {
...
+  after?: InputMaybe<Scalars['ID']['input']>;
+  filters?: InputMaybe<FilterGroup>;
+  first?: InputMaybe<Scalars['Int']['input']>;
+  orderBy?: InputMaybe<StixCoreObjectsOrdering>;
+  orderMode?: InputMaybe<OrderingMode>;
+  search?: InputMaybe<Scalars['String']['input']>;
+  types?: InputMaybe<Array<InputMaybe<Scalars['String']['input']>>>;
+};
+
+
```
Comment (aHenryJard) [inline opencti-platform/opencti-graphql/src/generated/graphql.ts:18225]:
```suggestion


```

File: opencti-platform/opencti-graphql/src/generated/graphql.ts [hunk 4]
```diff
@@ -35767,6 +35779,7 @@ export type QueryResolvers<ContextType = any, ParentType extends ResolversParent
   backgroundTask?: Resolver<Maybe<ResolversTypes['BackgroundTask']>, ParentType, ContextType, RequireFields<QueryBackgroundTaskArgs, 'id'>>;
   backgroundTasks?: Resolver<Maybe<ResolversTypes['BackgroundTaskConnection']>, ParentType, ContextType, Partial<QueryBackgroundTasksArgs>>;
   bookmarks?: Resolver<Maybe<ResolversTypes['StixDomainObjectConnection']>, ParentType, ContextType, Partial<QueryBookmarksArgs>>;
+  bulkSearch?: Resolver<Maybe<ResolversTypes['StixCoreObjectConnection']>, ParentType, ContextType, Partial<QueryBulkSearchArgs>>;
```
Comment (aHenryJard) [inline opencti-platform/opencti-graphql/src/generated/graphql.ts:35782]:
```suggestion
```

File: opencti-platform/opencti-graphql/src/config/providers.js [hunk 5]
```diff
@@ -152,6 +152,7 @@ for (let i = 0; i < providerKeys.length; i += 1) {
       providers.push({ name: providerName, type: AUTH_FORM, strategy, provider: 'local' });
     }
     if (strategy === STRATEGY_LDAP) {
+      logApp.info('[LDAP] LDAP strategy found in configuration.');
```
Comment (aHenryJard) [inline opencti-platform/opencti-graphql/src/config/providers.js]:
```suggestion
```

File: opencti-platform/opencti-graphql/config/schema/opencti.graphql [hunk 6]
```diff
@@ -11403,6 +11403,15 @@ type Query {
...
   ): StixCoreObjectConnection @auth(for: [KNOWLEDGE])
+  bulkSearch(
+    first: Int
+    after: ID
+    search: String
+    types: [String]
+    orderBy: StixCoreObjectsOrdering
+    orderMode: OrderingMode
+    filters: FilterGroup
+  ): StixCoreObjectConnection @auth(for: [KNOWLEDGE])
```
Comment (aHenryJard) [inline opencti-platform/opencti-graphql/config/schema/opencti.graphql]:
```suggestion
```

File: opencti-platform/opencti-graphql/src/generated/graphql.ts (hunk 4 above)
Comment (aHenryJard) [inline opencti-platform/opencti-graphql/src/generated/graphql.ts:35782]:
```suggestion
```

File: opencti-platform/opencti-front/src/private/components/common/bulk/dialog/BulkRelationDialog.tsx [hunk 7]
```diff
@@ -0,0 +1,374 @@
...
+          key: allEntitiesKeyList,
+          values: [text],
+        },
+      ],
+      filterGroups: [],
+    },
+    count: 1,
+  };
+
+  const result = await fetchQuery(
```
Comment (aHenryJard) [inline opencti-platform/opencti-front/src/private/components/common/bulk/dialog/BulkRelationDialog.tsx:121]: Usage of fetchQuery should be avoid in new code, we should use more usePreloadedQuery and loadQuery

Comment (aHenryJard) [reply opencti-platform/opencti-front/src/private/components/common/bulk/dialog/BulkRelationDialog.tsx:121]: no it's fine for now finally :D

File: opencti-platform/opencti-dev/2 - Frontend on test backend.run.xml [hunk 8]
```diff
@@ -1,12 +1,12 @@
 <component name="ProjectRunConfigurationManager">
-  <configuration default="false" name="2 - Frontend start cluster (second platform)" type="js.build_tools.npm">
+  <configuration default="false" name="2 - Frontend on test backend" type="js.build_tools.npm">
```
Comment (SouadHadjiat) [inline opencti-platform/opencti-dev/2 - Frontend on test backend.run.xml:2]: we don't have anymore "2 - Frontend start cluster (second platform).run.xml" ?

**Review by frapuks — APPROVED** [review]

File: opencti-platform/opencti-front/lang/front/de.json [hunk 9]
```diff
@@ -2115,6 +2121,7 @@
   "Representation entity": "Repräsentationsentität",
   "Representations for entity": "Darstellungen für Entität",
   "Representations for relationship": "Repräsentationen für Beziehung",
+  "Representative": "Repräsentativ",
```
Comment (SouadHadjiat) [inline opencti-platform/opencti-front/lang/front/de.json]: we used "representation" as a label for representative.

**Comment by SouadHadjiat** [comment]
Testing review
- **Missing some SDO (all SDO should be available for related to relationships)**
- Relationship types are only available in one direction (from intrusion set), we don't have the other direction relations => Not included in this POC

**Review by SouadHadjiat — APPROVED** [review]

**Pull request merged**